proxy/ParentConsistentHash.cc

```diff
@@ -63,21 +66,104 @@ ParentConsistentHash::getPathHash(HttpRequestData *hrdata, ATSHash64 *h)
 {
   const char *tmp = NULL;
   int len;
-  URL *url = hrdata->hdr->url_get();
+  URL *url     = hrdata->hdr->url_get();
+  int num_dirs = 0;
+
+  // Use over-ride URL from HttpTransact::State's cache_info.parent_selection_url, if present.
+  URL *ps_url = NULL;
+  Debug("parent_select", "hrdata->cache_info_parent_selection_url = %p", hrdata->cache_info_parent_selection_url);
+  if (hrdata->cache_info_parent_selection_url) {
+    ps_url = *(hrdata->cache_info_parent_selection_url);
+    Debug("parent_select", "ps_url = %p", ps_url);
+    if (ps_url) {
+      tmp = ps_url->string_get_ref(&len);
+      if (tmp && len > 0) {
+        // Print the over-ride URL
+        Debug("parent_select", "Using Over-Ride String='%.*s'.", len, tmp);
+        h->update(tmp, len);
+        h->final();
+        return h->get();
+      }
+    }
+  }
 
   // Always hash on '/' because paths returned by ATS are always stripped of it
   h->update("/", 1);
 
   tmp = url->path_get(&len);
-  if (tmp) {
+
+  if (tmp && len > 0) {
+    // Print the Original path.
+    Debug("parent_select", "Original Path='%.*s'.", len, tmp);
+
+    // Process the 'maxdirs' directive.
+    if (max_dirs != 0) {
+      // Determine number of directory components in the path.
+      // NOTE: Leading '/' is gone already.
+      for (int x = 0; x < len; x++) {
+        if (tmp[x] == '/')
+          num_dirs++;
```

Comment:
clang-format just formats. You can get clang-tidy to mostly do this for you if you want. Always using ``{}`` is the convention we would like to apply on all new code.